src/t8.c

```diff
@@ -34,7 +35,20 @@ t8_get_package_id (void)
 void
 t8_logv (int category, int priority, const char *fmt, va_list ap)
 {
-  sc_logv ("unknown", -1, t8_package_id, category, priority, fmt, ap);
+  char buffer[BUFSIZ];
+
+#ifdef SC_ENABLE_PTHREAD
+  sc_package_lock (t8_package_id);
+#endif
```

Comment:
Why do we need to lock this?
Furthermore, the `sc_package_lock` function does not need the compiler macros.